### PR TITLE
LPS-108588 DB Partition: Scheduled jobs

### PR DESCRIPTION
**Pull request by achaparro**
Hi @tinatian,

This pulls implements a framework to run appropriate scheduled jobs over every database partition in case this feature is enabled.

Tests passed successfully in @Preston-Crary and my repo:
https://github.com/Preston-Crary/liferay-portal/pull/1020#issuecomment-613941678
achaparro#554 (comment)

Thanks.

### CODE REVIEW

**Comment by achaparro** [comment]
ci:test:sf

**Comment by achaparro** [comment]
ci:test:relevant

File: modules/dxp/apps/saml/saml-impl/src/main/java/com/liferay/saml/runtime/internal/messaging/SamlMessageListener.java [hunk 1]
```diff
@@ -14,14 +14,15 @@
...
-import com.liferay.portal.kernel.messaging.BaseMessageListener;
+import com.liferay.portal.kernel.messaging.BaseMessageListenerCompanyScope;
 import com.liferay.portal.kernel.messaging.Message;
 import com.liferay.portal.kernel.messaging.MessageListenerException;
 
 /**
  * @author Tomas Polesovsky
  */
-public abstract class SamlMessageListener extends BaseMessageListener {
+public abstract class SamlMessageListener
```
Comment (tinatian) [inline modules/dxp/apps/saml/saml-impl/src/main/java/com/liferay/saml/runtime/internal/messaging/SamlMessageListener.java:24]: @achaparro  
Why this is a company scope listener? The impl is wrong, it is a loop.

SamlMessageListener#doReceive(message, long) -> BaseMessageListenerCompanyScope#doReceive(message) ->
SamlMessageListener#doReceive(message, long)

Comment (achaparro) [reply modules/dxp/apps/saml/saml-impl/src/main/java/com/liferay/saml/runtime/internal/messaging/SamlMessageListener.java:24]: @tinatian,
When we call this:
SamlMessageListener#doReceive(message, long) we will call the child implementation of doReceive(message) so it won't be an infinite loop

For SamlMetadataMessageListener we can keep the loop the by company when db partition is not enabled and for the other samllisteners, they will be executed once when db partition is not enabled and as many times as companies when db partition is enabled.

**Comment by achaparro** [comment]
I proceed to close this pull after talking to @tinatian. We will follow another approach wrapping the messagelistener to isolate the logic.

**Pull request closed without merging**